### DISCUSSION TITLE
How to change far clipping plane depth.

### DISCUSSION THREAD

**Discussion by olivia-banks** [gfx-rs/wgpu#2875] [Q&A] [answered]
Hello! I'm wondering how to change or set an arbitrary clipping plane depth. The only thing I found while searching online (as well as in the documentation) is implementing an [oblique view frustum](http://www.terathon.com/lengyel/Lengyel-Oblique.pdf). Given that wgpu already has a value set, I think it's plausible you can change it.

In short, polygons are clipped when far enough away from the "camera".

**Comment by kpreid** [comment, marked as answer]
I'm not sure I've understood your question, but normally, controlling the Z/depth clipping range is done by using via the projection matrix (or simpler arithmetic) in your vertex shader. Most graphics libraries that construct a projection matrix for you, will let you select the “near” and “far” distance to be clipped; these are the `znear` and `zfar` variables seen in [the learn-wgpu tutorial](https://sotrh.github.io/learn-wgpu/beginner/tutorial6-uniforms/#a-perspective-camera)).

If that doesn't help, can you clarify what you're looking for by providing the code you've got so far and maybe drawings of what difference you want?

**Reply by olivia-banks** [reply]
Sorry it wasn't that clear :/. I tried that, and nothing changed except introducing some weird visual artifacts. In hindsight, that probably should have told me I wasn't transferring or formatting some data correctly. I found out I was using incompatible types in the CPU and GPU representation.
Just fixed it and it works great! Sorry for wasting your time.
